src/components/settings/DeleteFormResponseModal/DeleteFormResponseModal.tsx

```diff
@@ -0,0 +1,101 @@
+import {
+  Banner,
+  Button,
+  Modal,
+  ModalFooter,
+  ModalHeader,
+} from "@/components/common";
+import { api } from "@convex/_generated/api";
+import type { Id } from "@convex/_generated/dataModel";
+import { useMutation } from "convex/react";
+import posthog from "posthog-js";
+import { useState } from "react";
+import type { Selection } from "react-aria-components";
+import { toast } from "sonner";
+import { getDeleteLabel } from "../FormResponsesList/FormResponsesList";
+
+interface DeleteFormResponseModalProps {
+  isOpen: boolean;
+  onOpenChange: (isOpen: boolean) => void;
+  onDelete: () => void;
+  selectedRows: Selection;
+  rows?: { id: Id<"userFormResponses">; field: string; value: string }[];
+}
+
+export function DeleteFormResponseModal({
+  isOpen,
+  onOpenChange,
+  onDelete,
+  selectedRows,
+  rows,
+}: DeleteFormResponseModalProps) {
+  const [isDeleting, setIsDeleting] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+
+  const deleteAll = useMutation(api.userFormResponses.deleteAll);
+  const deleteByIds = useMutation(api.userFormResponses.deleteByIds);
+
+  const deleteLabel = getDeleteLabel({
+    selectedRows,
+    totalCount: rows?.length ?? 0,
+  });
+
+  const handleDelete = async () => {
+    try {
+      setIsDeleting(true);
+
+      if (selectedRows === "all") {
+        await deleteAll();
+      } else {
+        await deleteByIds({
+          userFormResponseIds: Array.from(
+            selectedRows,
+          ) as Id<"userFormResponses">[],
+        });
+      }
+
+      onOpenChange(false);
+      onDelete();
+      toast.success("Data deleted");
```

Comment:
```suggestion
      toast.success("Response deleted");
```

Maybe for consistency?

Reply:
True. Changed the `getDeleteLabel` function to `getResponseCountLabel` so it can be reused to pluralize the confirmation here.